src/main/java/org/opensearch/security/action/apitokens/ApiToken.java

```diff
@@ -84,6 +95,38 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
             return builder;
         }
+
+        // TODO: look into integrating with default object mapper
+        @Override
```

Comment:
This is what `toXContent` is meant for. For objects that implement ToXContent you can jsonify them like this:

I believe you can do something like this to get json for debugging:

```
token.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS).toString();
```

Reply:
Nice! That's what I was looking for. I also implemented fromXcontent for the reverse (which I will need to use after decrypting the indexpermission from the token) in: [08e0890](https://github.com/opensearch-project/security/pull/4967/commits/08e08900b914f7df626692108b8099a745343bee)